components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/BPMNDataPublisher.java

```diff
@@ -152,15 +168,46 @@ public void configure() throws IOException, XMLStreamException, DataEndpointAuth
                         if ("true".equalsIgnoreCase(analyticsEnabledValue)) {
                             analyticsEnabled = true;
                         }
+                    } else if (beanProp.getAttributeValue(new QName(null, "name"))
+                            .equals(AnalyticsPublisherConstants.KPI_PUBLISHER_ENABLED_PROPERTY)) {
+                        String kpiAalyticsEnabledValue = beanProp.getAttributeValue(new QName(null, "value"));
+                        if ("true".equalsIgnoreCase(kpiAalyticsEnabledValue)) {
+                            kpiAnalyticsEnabled = true;
+                        }
                     }
                 }
             }
         }
 
         if (analyticsEnabled) {
+            configGenericVariablesPublishing(receiverURLSet, username, password, authURLSet, type);
+        }
+
+        if (kpiAnalyticsEnabled) {
+            configProcessVariablesPublishing(receiverURLSet, username, password, authURLSet, type);
+        }
+    }
+
+    /**
+     *
+     * @param receiverURLSet
+     * @param username
+     * @param password
+     * @param authURLSet
+     * @param type
+     * @throws DataEndpointAuthenticationException
+     * @throws DataEndpointAgentConfigurationException
+     * @throws TransportException
+     * @throws DataEndpointException
+     * @throws DataEndpointConfigurationException
+     */
```

Comment:
Seems this function is renamed/replaced later as configDataPublishing hence fine, but not description for parameters and exceptions